src/common/config.cpp

```diff
@@ -63,10 +58,31 @@ void Config::initialize(int argc, char** argv, cli::mode mode, bool validate) {
 
   log();
 
-  if(has("version"))
+  // Log version of Marian that has been used to create the model.
+  //
+  // Key "version" is present only if loaded from model parameters and is not
+  // related to --version flag
+  if(has("version")) {
+    auto version = get("version").as<std::string>();
+
+    if(version != PROJECT_VERSION_FULL)
+      LOG(info,
+          "[config] Loaded model has been created with Marian {}, "
+          "will be overwritten with current version {} at saving",
+          version,
+          PROJECT_VERSION_FULL);
+    else
+      LOG(info,
+          "[config] Loaded model has been created with Marian {}",
+          version);
+  }
+  // If this is a newly started training
+  else if(mode == cli::mode::training) {
     LOG(info,
-        "[config] Model created with Marian {}",
-        get("version").as<std::string>());
+        "[config] The model is being created with Marian {}",
+        PROJECT_VERSION_FULL);
+    config_["version"] = PROJECT_VERSION_FULL;
+  }
```

Comment:
I think we should log the Marian version every time, not only during training?

Reply:
It's also logged for translation with a Marian model. We have a regression test for this.

I see these cases: 
* (A) If a loaded model has been trained with Marian, the version will be loaded from the model configuration file and logged for training/translation/scoring.
* If there is no model configuration file, it means that it's either (B) a newly started training (message "Model is being created with...")
* or (C) translation/scoring with a model trained with Nematus or a very old Marian (no message).

"[config] version: xyz" will be displayed only if a version is loaded from the model configuration file.